app.js

```diff
@@ -7,42 +7,26 @@
   function gravatarURL(email = "", size = 160) {
     const emailMd5 = window.EagerGravatar.md5(email.toLowerCase())
 
-    return `http://www.gravatar.com/avatar/${emailMd5}.jpg?s=${size}`
+    return `https://www.gravatar.com/avatar/${emailMd5}.jpg?s=${size}`
   }
 
   function updateElement() {
     element = Eager.createElement(options.location, element)
     element.classList.add(CONTAINER_CLASS)
 
-    const imageSrc = options.image ? options.image : gravatarURL(options.email)
-    const wrapper = document.createElement("eager-wrapper")
+    element.setAttribute("data-position", options.position)
 
-    // TODO: something like
-    wrapper.setAttribute("data-position", options.position)
+    const {profilePicture} = options
+    const imageSrc = profilePicture.type === "upload" && profilePicture.url || gravatarURL(profilePicture.gravatarEmail)
+    const wrapper = document.createElement("eager-wrapper")
 
     wrapper.innerHTML = `
       <eager-details>
-        <eager-avatar style="background-image: url(${imageSrc});"></eager-avatar>
+        <eager-profile-picture style="background-image: url(${imageSrc});"></eager-profile-picture>
         <eager-message>${options.message.html}</eager-message>
       </eager-details>
     `
 
-    if (options.email) {
```

Comment:
In an effort to lower the amount of options and responsibilities, the social icons have been removed. This is better handled in a [dedicated app](https://eager.io/app/social-icons).